### DISCUSSION TITLE
CONTRIBUTING の整理

### DISCUSSION THREAD

**Discussion by gurezo** [chirimen-oh/meeting#35] [General]
## 目的

- 参加した人が CONTRIBUTE し易いように導線を整理したい

## ゴール

- 各リポジトリの CONTRIBUTE への導線を記載する

## 参考
- ~~chirimen-drivers の[CONTRIBUTING リンク切れ対応 #297](https://github.com/chirimen-oh/chirimen-drivers/issues/297)~~
  - ~~上記 issue は暫定処置~~ done

## 議論

- CONTRIBUTE のルールはどうするか？
- CONTRIBUTE への導線をどうするか？
  - 案1：一箇所にまとめる
  - 案2：各リポジトリ毎に記載する
- issue について
  - 以下のものは各リポジトリに issue を立てるという事で良いか？
    - https://tutorial.chirimen.org/
    - https://github.com/chirimen-oh/chirimen-lite
    - https://github.com/chirimen-oh/chirimen-drivers
    - https://github.com/chirimen-oh/node-web-gpio
    - https://github.com/chirimen-oh/node-web-i2c
    - https://github.com/chirimen-oh/chirimen
- [pre-arrangement-contributions](https://github.com/chirimen-oh/pre-arrangement-contributions)とのCONTRIBUTINGの棲み分け

**Comment by gurezo** [comment]
[pre-arrangement-contributions](https://github.com/chirimen-oh/pre-arrangement-contributions)とのCONTRIBUTINGの棲み分け は、 @satakagi さんを交えて

**Comment by kou029w** [comment]
各リポジトリごとに目的異なるかと思いますので各リポジトリごとで良いかと思います > CONTRIBUTING
（それらを chirimen.org でリンク一覧をまとめる、あるいは相互リンク加えるなどあるとより良いのかとおもいました）

**Comment by gurezo** [comment]
## gurezo/chirimen-contributing-guide

- https://github.com/gurezo/chirimen-contributing-guide

---

### Overview

`gurezo/chirimen-contributing-guide` で整理した内容を、将来的に `chirimen-oh` Organization へどのように移管・統合するか提案をまとめます。

### Background

このリポジトリは、最初から独立した恒久プロジェクトとすることを目的としていません。

まず `gurezo` 配下で既存の暗黙知を整理し、コミュニティでレビュー可能な状態を作ったうえで、正式な配置を検討します。

### Options

少なくとも以下を比較します。

#### Option A: Repository transfer

`chirimen-contributing-guide` 自体を `chirimen-oh` Organization に移管する。

#### Option B: Split and integrate

共通情報と Repository 固有情報を分割し、それぞれ既存 Repository に統合する。

例:

```text
共通情報
  ↓
Organization-level contribution guide

Repository 固有情報
  ↓
各 Repository の CONTRIBUTING.md / docs
```

#### Option C: Hybrid

Organization 共通ガイドを残しつつ、詳細手順を各 Repository 側で管理する。

### Considerations

- Single source of truth
- メンテナンス責任
- Repository 固有情報との重複
- Wiki との役割分担
- Discussions との役割分担
- Webサイトからの導線
- 初心者から見た分かりやすさ
- 長期的な更新のしやすさ

### Deliverable

CHIRIMEN Organization Discussions でレビュー可能な Migration / Integration Proposal を作成します。

### Definition of Done

- [ ] 移管・統合方法の候補が整理されている
- [ ] 各候補のメリット・課題が整理されている
- [ ] 推奨する構成案ではなく、コミュニティが判断できる材料が揃っている
- [ ] Organization Discussions に提案できる文章になっている
